### PR TITLE
Fix missing pantherlog source fields

### LINKED ISSUE #2180: Some log type reporting null p_source_id, p_source_label
### Describe the bug

Seeing null `p_source_id`, `p_source_label` columns for data onboarding through S3.

### Steps to reproduce

1. Go to Log Analysis -> Sources
2. Onboard a CloudTrail log type
3. Wait until you have received some data
4. Run the query :
```
SELECT p_source_label, p_source_id FROM panther_logs.aws_cloudtrail
```
5. See `NULL`s

### Expected behavior

`p_source_id`, `p_source_label` should ALWAYS be populated

### Environment

How are you deploying or using Panther?

- Panther version or commit: 1.13

### Additional context

This seems to affect data that have been migrated to the new parser scheme

### PR DESCRIPTION
**Pull request by alxarch**
## Background

Closes #2180 

## Changes

- Add missing fields when encoding `pantherlog.Result` to JSON

## Testing

- mage test:go

### CODE REVIEW

File: internal/log_analysis/log_processor/pantherlog/jsoniter.go [hunk 1]
```diff
@@ -129,6 +129,18 @@ func (*resultEncoder) writePantherFields(r *Result, stream *jsoniter.Stream) {
 	stream.WriteObjectField(FieldParseTimeJSON)
 	stream.WriteVal(r.PantherParseTime)
 
+	if r.PantherSourceID != "" {
```
Comment (kostaspap) [inline internal/log_analysis/log_processor/pantherlog/jsoniter.go:132]: Do we need such check? can these ever be null? 
Better fail early?

File: results.jsonl [hunk 2]
```diff
@@ -0,0 +1,5 @@
...4-6ad9-4448-8b5e-cef6a627301c","eventName":"Decrypt","eventSource":"kms.amazonaws.com","eventTime":"2020-11-20 16:19:52.000000000","eventType":"AwsApiCall","eventVersion":"1.05","readOnly":true,"recipientAccountId":"050603629990","requestID":"9c39b5cd-6702-4703-a9d9-5a334efeca5b","requestParameters":{"encryptionAlgorithm":"SYMMETRIC_DEFAULT","encryptionContext":{"aws:dynamodb:tableName":"panther-source-integrations","aws:dynamodb:subscriberId":"050603629990"}},"resources":[{"arn":"arn:aws:kms:us-east-1:050603629990:key/30fbc245-d530-436b-8131-1082b6091361","accountId":"050603629990","type":"AWS::KMS::Key"}],"responseElements":null,"sharedEventID":"f33e3895-ad97-4eb5-a80f-4e5465d6b11b","sourceIPAddress":"dynamodb.amazonaws.com","userAgent":"dynamodb.amazonaws.com","userIdentity":{"type":"AWSService","invokedBy":"dynamodb.amazonaws.com"},"p_log_type":"AWS.CloudTrail","p_row_id":"002b673d3ec7e4dbb8f5a7d00601","p_event_time":"2020-11-20 16:19:52.000000000","p_parse_time":"2020-12-03 08:36:13.455650469","p_any_aws_account_ids":["050603629990"],"p_any_aws_arns":["arn:aws:kms:us-east-1:050603629990:key/30fbc245-d530-436b-8131-1082b6091361"],"p_any_domain_names":["dynamodb.amazonaws.com"]}
```
Comment (kostaspap) [inline results.jsonl:1]: Should be removed?

**Comment by kostaspap** [comment]
I would recommend adding the `sync-to-release` label

File: internal/log_analysis/log_processor/pantherlog/jsoniter.go (hunk 1 above)
Comment (alxarch) [reply internal/log_analysis/log_processor/pantherlog/jsoniter.go:132]: This is the equivalent of `omitempty`. This part of the code is the customized  `JSON` encoder for a `pantherlog.Result`. It does not care about where this is used (i.e. log processor, CLI, tests). Source ID is only present in the log processor.

File: results.jsonl (hunk 2 above)
Comment (alxarch) [reply results.jsonl]: fixed

**Review by kostaspap — APPROVED** [review]

**Pull request merged**